src/NuGet.Core/NuGet.Commands/RestoreCommand/ToolRestoreResult.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using NuGet.ProjectModel;
+
+namespace NuGet.Commands
+{
+    public class ToolRestoreResult
```

Comment:
I decided not to add it here, since there is currently no need for it. A tool restore cannot happen without full project restore so the success is only tracking on the `RestoreResult` (which has many `ToolRestoreResult`).